### DISCUSSION TITLE
Project Body Bike

### DISCUSSION THREAD

**Discussion by timo-r3** [doudar/SmartSpin2k#199] [Show and tell]
Hi! Here is my project so far. My spinnig bike is oldish Body Bike (https://body-bike.com/). The insert for my knob which @doudar made is perfecet, thank you!
I did not find anybody to print the parts so I bought Creality Ender 3 pro. First everything went fine, but the case did not succeed, I don't know what is wrong, maybe some setting or need to level the bed again. I use PETG filament:
![insert](https://user-images.githubusercontent.com/37155777/116002327-f0cbd380-a601-11eb-871f-689abd658aa6.jpg)
![knob_cup](https://user-images.githubusercontent.com/37155777/116002328-f2959700-a601-11eb-862c-a4adab7e229d.jpg)
![gear 30T](https://user-images.githubusercontent.com/37155777/116002331-f4f7f100-a601-11eb-921c-646507d1fc79.jpg)
![gear 11T](https://user-images.githubusercontent.com/37155777/116002333-f6c1b480-a601-11eb-8ce0-51e15e787f4c.jpg)
![bottom case](https://user-images.githubusercontent.com/37155777/116002335-f88b7800-a601-11eb-82d8-ddbf8e31e02a.jpg)
![bottom case 2](https://user-images.githubusercontent.com/37155777/116002341-faedd200-a601-11eb-8edf-5613345e8b8f.jpg)

Wiring went pretty straight forward. Is the LM2596 used to provide +5V both to the ESP32 and TMC2225? That's how I wired it. I use laptop PSU and it gives +19V, that is connected the to the LM2596 and TMC2225. So TMC gets both +19V and +5V in? There could be easier wiring diagram for newbies like me :-) Here  are some pictures:
![wiring1](https://user-images.githubusercontent.com/37155777/116002507-c2022d00-a602-11eb-9f17-a6584b390da0.jpg)
![wiring2](https://user-images.githubusercontent.com/37155777/116002508-c4648700-a602-11eb-86e3-8a2f4dde7887.jpg)
I plan to solder PSU cables to LM2596 

I managed to upload the initial firmware to the ESP32 and after setting wifi connection, it downloaded newest firmware. I used the failed case, it okayish when I removed all the filament strings. I need to print new case sometime later. Up and down shiting works fine. I use Assioma power pedals. I managed to connect them. There was maybe something unclear in the pairing process which I don't remeber anymore, but now it works just fine every time. Today I made initial test with Sufferfest, and it seems to work fine. Only thing I wonder, how fast SmartSpin2k will react to power changes, now it feels quite long before the resistance is altered, it makes so small steps. But I need to test that more.
![sufferfest](https://user-images.githubusercontent.com/37155777/116002989-d810ed00-a604-11eb-998e-f0ba0354bed5.jpg)

My biggest challenge know is to make good printouts for the case and finalize the cabling. Nice project!

**Comment by doudar** [comment]
Looking good! (Except that case lol!)

Post your Cura settings and maybe I can give you some hints. I'm guessing you may have gotten a nozzle blob or possibly an extruder skip which wrecked the rest of the print. There's lots of retractions on the case so if those aren't dialed in correctly, that could be cause an issue.

You could try PLA for the case if you have a spool. It might not last years out of PLA like it will in PETG, but it's more forgiving for settings. 

When you put the small gear on the stepper shaft, it helps if you heat it up slightly with a heat gun/hair dryer. It needs to fit tight because it doesn't have any other retention.

Following along here - good luck!

Also, if you come up with any changes/additions to the wiring diagrams, please feel free to share them!

**Reply by doudar** [reply]
We've also got some tweaks upstream in the erg code that should be merged into Develop soon. 

In the meantime, you could make erg mode more aggressive by increasing your shift step size.
